lib/connection_string.js

```diff
@@ -500,6 +493,46 @@ function checkTLSOptions(queryString) {
   }
 }
 
+/**
+ * Checks options object if both options are present (any value) will throw an error.
+ *
+ * @param {Object} options Options Object
+ * @param {string} optionKeyA A options key
+ * @param {string} optionKeyB B options key
+ * @throws {MongoParseError}
+ */
+function assertRepelOptions(options, optionKeyA, optionKeyB) {
+  if (options.hasOwnProperty(optionKeyA) && options.hasOwnProperty(optionKeyB)) {
+    throw new MongoParseError(`The \`${optionKeyA}\` option cannot be used with \`${optionKeyB}\``);
+  }
+}
```

Comment:
💡 Just an idea: This function is rather generic maybe it is worthy of being a util?

Reply:
There's already another `assert` function in here, I'd hold off on contaminating more files.

Reply:
Not implementing, is that OK